hadoop-tools/hadoop-distcp/src/main/java/org/apache/hadoop/tools/mapred/RetriableFileCopyCommand.java

```diff
@@ -247,6 +276,22 @@ private long copyToFile(Path targetPath, FileSystem targetFS,
         context);
   }
 
+  private InetSocketAddress[] toFavoredNodes(String favoredNodesStr) throws UnknownHostException {
+    List<InetSocketAddress> result = new ArrayList<>();
+    for (String hostAndPort : favoredNodesStr.split(",")) {
```

Comment:
what happens if an empty string is passed in? it should be an error, right? so add a test

Reply:
I have checked if `favoredNodesStr ` is an empty string above, and will skip `favoredNodesStr` if empty.